lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -1236,4 +1236,54 @@ private function isNumericType(Type $type) : bool
     {
         return $type instanceof IntegerType || $type instanceof BigIntType;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getDropIndexSQL($index, $table = null)
+    {
+        $isUnique = false;
+        if ($index instanceof Index) {
+            $isUnique = $index->isUnique();
+            $index = $index->getQuotedName($this);
+        } elseif (!is_string($index)) {
+            throw new \InvalidArgumentException('AbstractPlatform::getDropIndexSQL() expects $index parameter to be string or \Doctrine\DBAL\Schema\Index.');
+        }
+
+        if ($isUnique && $table) {
+            return 'ALTER TABLE ' . $table . ' DROP CONSTRAINT ' . $index;
+        } else {
+            return 'DROP INDEX ' . $index;
```

Comment:
If a unique constraint was created using `ALTER TABLE […] ADD CONSTRAINT` and now is being dropped using `DROP INDEX`, will it succeed? If not, I'd say using indices instead of constraints in DBAL is by design since it only operates indices.

Reply:
_Backward-Remains-Compatible_:

- All operations involve `<indexes>` and `<index>`.
- `<uniqueConstraints>`: Renaming is ok. `<uniqueConstraint name="old_name_constraints" columns="..." />` to `<uniqueConstraint name="new_name_constraints" />`
- Freshly added  `<uniqueConstraints>`

_Things that will BC-Break_ (although won't break production, will likely break their builds):

- Existing schema that uses `<uniqueConstraints>` generated with `CREATE UNIQUE INDEX indexName ON table (...)` will not be able to be dropped. Those can only be dropped using old syntax `DROP INDEX indexName`, but dbal will generate `ALTER TABLE DROP CONSTRAINT indexName`

Admittedly this could be totally prevented with better implementation.

Reply:
context: `drop index` on constraints created using `ALTER TABLE`

`tablename=# DROP INDEX unique_constraint_name;`
`ERROR:  cannot drop index unique_constraint_name because constraint unique_constraint_name on table tablename requires it`
`HINT:  You can drop constraint unique_constraint_name on table tablename instead.`

However: renaming is perfectly fine. In fact, the renaming syntax is not changed and continue to use the original syntax.

Reply:
context:

Dropping index using `DROP INDEX constraintname` ON constraint created using `ALTER TABLE tablename ADD CONSTRAINT constraintname UNQIUE(...)` will encounter error `constraint indexName not found on table`

Vice versa:

Dropping index using `ALTER TABLE tablename DROP CONSTRAINT constraintname` on constraint created using `CREATE INDEX UNIQUE constraintname...` will encounter error `... table requires it, you can drop constraint constraintname on table tablename instead`

Let me find out if schemaManager provides constraint information on non-foreignKey constraint...